python/ray/data/_internal/planner/plan_udf_map_op.py

```diff
@@ -366,16 +368,19 @@ async def process_batch(batch: DataBatch):
                 )  # Put the exception into the queue to signal an error
 
         async def process_all_batches():
-            loop = ray.data._map_actor_context.udf_map_asyncio_loop
-            tasks = [loop.create_task(process_batch(x)) for x in input_iterable]
+            try:
+                loop = ray.data._map_actor_context.udf_map_asyncio_loop
```

Comment:
Out of scope for this PR, but do you know why we need to use a global variable for the loop? Seems like the actor context is specific to a UDF?

Reply:
I guess it's because we already have a global var `_MapActorContext` to cache other stuff. 
technically the loop can also be bound to the actor object.